daemon-scheduler/internal/features/wrappers/autoscaling_wrapper.go

```diff
@@ -44,3 +51,94 @@ func (wrapper AutoScalingWrapper) SetDesiredCapacity(asg string, count int64) er
 	}
 	return nil
 }
+
+func (wrapper AutoScalingWrapper) CreateAutoScalingGroup(asg string, configName string, availabilityZones []*string) error {
+	minSize := int64(0)
+	// MaxSize is set to 10 because it's the maximum number of instances launched by the e2e test
+	maxSize := int64(10)
+	in := autoscaling.CreateAutoScalingGroupInput{
+		AutoScalingGroupName: aws.String(asg),
+		MinSize: aws.Int64(minSize),
+		MaxSize: aws.Int64(maxSize),
+		LaunchConfigurationName: aws.String(configName),
+		AvailabilityZones: availabilityZones,
+	}
+
+	_, err := wrapper.client.CreateAutoScalingGroup(&in)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (wrapper AutoScalingWrapper) DeleteAutoScalingGroup(asg string, forceDelete bool) error {
+	in := autoscaling.DeleteAutoScalingGroupInput{
+		AutoScalingGroupName: aws.String(asg),
+		ForceDelete: aws.Bool(forceDelete),
+	}
+
+	_, err := wrapper.client.DeleteAutoScalingGroup(&in)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (wrapper AutoScalingWrapper) CreateLaunchConfiguration(configName string, clusterName string, instanceProfileName string, keyName string, amiID string)  error {
+	// The user data is run when the instance is launched. It registers the instance to the cluster
+	// and makes the instance to self-terminate after one hour in case the test fails to clean up.
```

Comment:
Does self-terminate actually work? If not, might make sense to point out that the instance state is set to stopped (and that it does not actually terminate).